From 411af99bdf6882100395d8cd16f5e42e68c62b4a Mon Sep 17 00:00:00 2001
From: songliyong <songliyong@contributor>


Date: Thu, 10 Apr 2025 10:27:34 +0800
Subject: [PATCH] USERSTORY fix threadpool related mtr testcases

Issue:
------
Threadpool add some info in information_schema, so need to update realdated testcases

Solution:
---------
Involved testcases:
is_tables_is.test
is_columns_is_cs.test
information_schema_db.test
---
 .../suite/funcs_1/r/is_columns_is_cs.result   |  90 +++++++++
 .../suite/funcs_1/r/is_tables_is.result       | 184 ++++++++++++++++++
 .../r/information_schema_db.result            |   8 +
 3 files changed, 282 insertions(+)

diff --git a/mysql-test/suite/funcs_1/r/is_columns_is_cs.result b/mysql-test/suite/funcs_1/r/is_columns_is_cs.result
index 3739c05ea54..cbc272c94e7 100644
--- a/mysql-test/suite/funcs_1/r/is_columns_is_cs.result
+++ b/mysql-test/suite/funcs_1/r/is_columns_is_cs.result
@@ -413,6 +413,51 @@ def	information_schema	TABLE_PRIVILEGES	TABLE_SCHEMA	3		NO	varchar	21	64	NULL	NU
 def	information_schema	TABLE_PRIVILEGES	TABLE_NAME	4		NO	varchar	21	64	NULL	NULL	NULL	utf8mb3	utf8mb3_general_ci	varchar(64)			select			NULL
 def	information_schema	TABLE_PRIVILEGES	PRIVILEGE_TYPE	5		NO	varchar	21	64	NULL	NULL	NULL	utf8mb3	utf8mb3_general_ci	varchar(64)			select			NULL
 def	information_schema	TABLE_PRIVILEGES	IS_GRANTABLE	6		NO	varchar	1	3	NULL	NULL	NULL	utf8mb3	utf8mb3_general_ci	varchar(3)			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	GROUP_ID	1		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	CONNECTIONS	2		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	THREADS	3		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	ACTIVE_THREADS	4		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	STANDBY_THREADS	5		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	LOWPRIO_QUEUE_LENGTH	6		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	HIGHPRIO_QUEUE_LENGTH	7		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	HAS_LISTENER	8		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	IS_STALLED	9		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	THREADPOOL_SIZE	10		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	CPU_COUNT	11		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	CPU_TIME	12		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	WAIT_TIME	13		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	DEQUEUED EVENTS	14		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	EVENT_COUNT	15		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_GROUPS	IO_EVENT_COUNT	16		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_QUEUES	GROUP_ID	1		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_QUEUES	POSITION	2		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_QUEUES	PRIORITY	3		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_QUEUES	CONNECTION_ID	4		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int unsigned			select			NULL
+def	information_schema	THREAD_POOL_QUEUES	QUEUEING_TIME_MICROSECONDS	5		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int unsigned			select			NULL
+def	information_schema	THREAD_POOL_QUEUES	QUEUEING_HTIME_MICROSECONDS	6		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int unsigned			select			NULL
+def	information_schema	THREAD_POOL_QUEUES	TICKETS	7		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint unsigned			select			NULL
+def	information_schema	THREAD_POOL_QUEUES	KICKUP	8		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_STATS	GROUP_ID	1		NO	int	NULL	NULL	NULL	NULL	NULL	NULL	NULL	int			select			NULL
+def	information_schema	THREAD_POOL_STATS	THREAD_CREATIONS	2		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	THREAD_CREATIONS_DUE_TO_STALL	3		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	WAKES	4		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	WAKES_DUE_TO_STALL	5		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	THROTTLES	6		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	STALLS	7		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	POLLS_BY_LISTENER	8		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	POLLS_BY_WORKER	9		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	DEQUEUES_BY_LISTENER	10		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	DEQUEUES_BY_WORKER	11		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	KICKUPS	12		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	THREAD_CREATIONS_QUEUE_PUT_CONNECTION	13		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	THREAD_CREATIONS_WAIT_BEGIN	14		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	THREAD_CREATIONS_SINGLE_LISTENER	15		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	THREAD_CREATION_ATTEMPTS	16		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	THREAD_REMOVES	17		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	CPU_TIMER	18		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_STATS	EVENTS_HANDLED_CUR	19		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
+def	information_schema	THREAD_POOL_WAITS	REASON	1		NO	varchar	5	16	NULL	NULL	NULL	utf8mb3	utf8mb3_general_ci	varchar(16)			select			NULL
+def	information_schema	THREAD_POOL_WAITS	COUNT	2		NO	bigint	NULL	NULL	NULL	NULL	NULL	NULL	NULL	bigint			select			NULL
 def	information_schema	TRIGGERS	TRIGGER_CATALOG	1	NULL	NO	varchar	64	192	NULL	NULL	NULL	utf8mb3	utf8mb3_bin	varchar(64)			select			NULL
 def	information_schema	TRIGGERS	TRIGGER_SCHEMA	2	NULL	NO	varchar	64	192	NULL	NULL	NULL	utf8mb3	utf8mb3_bin	varchar(64)			select			NULL
 def	information_schema	TRIGGERS	TRIGGER_NAME	3	NULL	NO	varchar	64	192	NULL	NULL	NULL	utf8mb3	utf8mb3_general_ci	varchar(64)			select			NULL
@@ -957,6 +1002,51 @@ NULL	information_schema	TABLE_CONSTRAINTS_EXTENSIONS	SECONDARY_ENGINE_ATTRIBUTE
 3.0476	information_schema	TABLE_PRIVILEGES	TABLE_NAME	varchar	21	64	utf8mb3	utf8mb3_general_ci	varchar(64)
 3.0476	information_schema	TABLE_PRIVILEGES	PRIVILEGE_TYPE	varchar	21	64	utf8mb3	utf8mb3_general_ci	varchar(64)
 3.0000	information_schema	TABLE_PRIVILEGES	IS_GRANTABLE	varchar	1	3	utf8mb3	utf8mb3_general_ci	varchar(3)
+NULL	information_schema	THREAD_POOL_GROUPS	GROUP_ID	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	CONNECTIONS	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	THREADS	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	ACTIVE_THREADS	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	STANDBY_THREADS	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	LOWPRIO_QUEUE_LENGTH	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	HIGHPRIO_QUEUE_LENGTH	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	HAS_LISTENER	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	IS_STALLED	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	THREADPOOL_SIZE	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	CPU_COUNT	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	CPU_TIME	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	WAIT_TIME	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	DEQUEUED EVENTS	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	EVENT_COUNT	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_GROUPS	IO_EVENT_COUNT	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_QUEUES	GROUP_ID	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_QUEUES	POSITION	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_QUEUES	PRIORITY	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_QUEUES	CONNECTION_ID	int	NULL	NULL	NULL	NULL	int unsigned
+NULL	information_schema	THREAD_POOL_QUEUES	QUEUEING_TIME_MICROSECONDS	int	NULL	NULL	NULL	NULL	int unsigned
+NULL	information_schema	THREAD_POOL_QUEUES	QUEUEING_HTIME_MICROSECONDS	int	NULL	NULL	NULL	NULL	int unsigned
+NULL	information_schema	THREAD_POOL_QUEUES	TICKETS	bigint	NULL	NULL	NULL	NULL	bigint unsigned
+NULL	information_schema	THREAD_POOL_QUEUES	KICKUP	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_STATS	GROUP_ID	int	NULL	NULL	NULL	NULL	int
+NULL	information_schema	THREAD_POOL_STATS	THREAD_CREATIONS	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	THREAD_CREATIONS_DUE_TO_STALL	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	WAKES	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	WAKES_DUE_TO_STALL	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	THROTTLES	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	STALLS	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	POLLS_BY_LISTENER	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	POLLS_BY_WORKER	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	DEQUEUES_BY_LISTENER	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	DEQUEUES_BY_WORKER	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	KICKUPS	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	THREAD_CREATIONS_QUEUE_PUT_CONNECTION	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	THREAD_CREATIONS_WAIT_BEGIN	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	THREAD_CREATIONS_SINGLE_LISTENER	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	THREAD_CREATION_ATTEMPTS	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	THREAD_REMOVES	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	CPU_TIMER	bigint	NULL	NULL	NULL	NULL	bigint
+NULL	information_schema	THREAD_POOL_STATS	EVENTS_HANDLED_CUR	bigint	NULL	NULL	NULL	NULL	bigint
+3.2000	information_schema	THREAD_POOL_WAITS	REASON	varchar	5	16	utf8mb3	utf8mb3_general_ci	varchar(16)
+NULL	information_schema	THREAD_POOL_WAITS	COUNT	bigint	NULL	NULL	NULL	NULL	bigint
 3.0000	information_schema	TRIGGERS	TRIGGER_CATALOG	varchar	64	192	utf8mb3	utf8mb3_bin	varchar(64)
 3.0000	information_schema	TRIGGERS	TRIGGER_SCHEMA	varchar	64	192	utf8mb3	utf8mb3_bin	varchar(64)
 3.0000	information_schema	TRIGGERS	TRIGGER_NAME	varchar	64	192	utf8mb3	utf8mb3_general_ci	varchar(64)
diff --git a/mysql-test/suite/funcs_1/r/is_tables_is.result b/mysql-test/suite/funcs_1/r/is_tables_is.result
index b27c12ce487..c8bfe550296 100644
--- a/mysql-test/suite/funcs_1/r/is_tables_is.result
+++ b/mysql-test/suite/funcs_1/r/is_tables_is.result
@@ -960,6 +960,98 @@ user_comment
 Separator	-----------------------------------------------------
 TABLE_CATALOG	def
 TABLE_SCHEMA	information_schema
+TABLE_NAME	THREAD_POOL_GROUPS
+TABLE_TYPE	SYSTEM VIEW
+ENGINE	NULL
+VERSION	10
+ROW_FORMAT	NULL
+TABLE_ROWS	#TBLR#
+AVG_ROW_LENGTH	#ARL#
+DATA_LENGTH	#DL#
+MAX_DATA_LENGTH	#MDL#
+INDEX_LENGTH	#IL#
+DATA_FREE	#DF#
+AUTO_INCREMENT	#AI#
+CREATE_TIME	#CRT#
+UPDATE_TIME	#UT#
+CHECK_TIME	#CT#
+TABLE_COLLATION	NULL
+CHECKSUM	NULL
+CREATE_OPTIONS	#CO#
+TABLE_COMMENT	#TC#
+user_comment	
+Separator	-----------------------------------------------------
+TABLE_CATALOG	def
+TABLE_SCHEMA	information_schema
+TABLE_NAME	THREAD_POOL_QUEUES
+TABLE_TYPE	SYSTEM VIEW
+ENGINE	NULL
+VERSION	10
+ROW_FORMAT	NULL
+TABLE_ROWS	#TBLR#
+AVG_ROW_LENGTH	#ARL#
+DATA_LENGTH	#DL#
+MAX_DATA_LENGTH	#MDL#
+INDEX_LENGTH	#IL#
+DATA_FREE	#DF#
+AUTO_INCREMENT	#AI#
+CREATE_TIME	#CRT#
+UPDATE_TIME	#UT#
+CHECK_TIME	#CT#
+TABLE_COLLATION	NULL
+CHECKSUM	NULL
+CREATE_OPTIONS	#CO#
+TABLE_COMMENT	#TC#
+user_comment	
+Separator	-----------------------------------------------------
+TABLE_CATALOG	def
+TABLE_SCHEMA	information_schema
+TABLE_NAME	THREAD_POOL_STATS
+TABLE_TYPE	SYSTEM VIEW
+ENGINE	NULL
+VERSION	10
+ROW_FORMAT	NULL
+TABLE_ROWS	#TBLR#
+AVG_ROW_LENGTH	#ARL#
+DATA_LENGTH	#DL#
+MAX_DATA_LENGTH	#MDL#
+INDEX_LENGTH	#IL#
+DATA_FREE	#DF#
+AUTO_INCREMENT	#AI#
+CREATE_TIME	#CRT#
+UPDATE_TIME	#UT#
+CHECK_TIME	#CT#
+TABLE_COLLATION	NULL
+CHECKSUM	NULL
+CREATE_OPTIONS	#CO#
+TABLE_COMMENT	#TC#
+user_comment	
+Separator	-----------------------------------------------------
+TABLE_CATALOG	def
+TABLE_SCHEMA	information_schema
+TABLE_NAME	THREAD_POOL_WAITS
+TABLE_TYPE	SYSTEM VIEW
+ENGINE	NULL
+VERSION	10
+ROW_FORMAT	NULL
+TABLE_ROWS	#TBLR#
+AVG_ROW_LENGTH	#ARL#
+DATA_LENGTH	#DL#
+MAX_DATA_LENGTH	#MDL#
+INDEX_LENGTH	#IL#
+DATA_FREE	#DF#
+AUTO_INCREMENT	#AI#
+CREATE_TIME	#CRT#
+UPDATE_TIME	#UT#
+CHECK_TIME	#CT#
+TABLE_COLLATION	NULL
+CHECKSUM	NULL
+CREATE_OPTIONS	#CO#
+TABLE_COMMENT	#TC#
+user_comment	
+Separator	-----------------------------------------------------
+TABLE_CATALOG	def
+TABLE_SCHEMA	information_schema
 TABLE_NAME	TRIGGERS
 TABLE_TYPE	SYSTEM VIEW
 ENGINE	NULL
@@ -2060,6 +2152,98 @@ user_comment
 Separator	-----------------------------------------------------
 TABLE_CATALOG	def
 TABLE_SCHEMA	information_schema
+TABLE_NAME	THREAD_POOL_GROUPS
+TABLE_TYPE	SYSTEM VIEW
+ENGINE	NULL
+VERSION	10
+ROW_FORMAT	NULL
+TABLE_ROWS	#TBLR#
+AVG_ROW_LENGTH	#ARL#
+DATA_LENGTH	#DL#
+MAX_DATA_LENGTH	#MDL#
+INDEX_LENGTH	#IL#
+DATA_FREE	#DF#
+AUTO_INCREMENT	#AI#
+CREATE_TIME	#CRT#
+UPDATE_TIME	#UT#
+CHECK_TIME	#CT#
+TABLE_COLLATION	NULL
+CHECKSUM	NULL
+CREATE_OPTIONS	#CO#
+TABLE_COMMENT	#TC#
+user_comment	
+Separator	-----------------------------------------------------
+TABLE_CATALOG	def
+TABLE_SCHEMA	information_schema
+TABLE_NAME	THREAD_POOL_QUEUES
+TABLE_TYPE	SYSTEM VIEW
+ENGINE	NULL
+VERSION	10
+ROW_FORMAT	NULL
+TABLE_ROWS	#TBLR#
+AVG_ROW_LENGTH	#ARL#
+DATA_LENGTH	#DL#
+MAX_DATA_LENGTH	#MDL#
+INDEX_LENGTH	#IL#
+DATA_FREE	#DF#
+AUTO_INCREMENT	#AI#
+CREATE_TIME	#CRT#
+UPDATE_TIME	#UT#
+CHECK_TIME	#CT#
+TABLE_COLLATION	NULL
+CHECKSUM	NULL
+CREATE_OPTIONS	#CO#
+TABLE_COMMENT	#TC#
+user_comment	
+Separator	-----------------------------------------------------
+TABLE_CATALOG	def
+TABLE_SCHEMA	information_schema
+TABLE_NAME	THREAD_POOL_STATS
+TABLE_TYPE	SYSTEM VIEW
+ENGINE	NULL
+VERSION	10
+ROW_FORMAT	NULL
+TABLE_ROWS	#TBLR#
+AVG_ROW_LENGTH	#ARL#
+DATA_LENGTH	#DL#
+MAX_DATA_LENGTH	#MDL#
+INDEX_LENGTH	#IL#
+DATA_FREE	#DF#
+AUTO_INCREMENT	#AI#
+CREATE_TIME	#CRT#
+UPDATE_TIME	#UT#
+CHECK_TIME	#CT#
+TABLE_COLLATION	NULL
+CHECKSUM	NULL
+CREATE_OPTIONS	#CO#
+TABLE_COMMENT	#TC#
+user_comment	
+Separator	-----------------------------------------------------
+TABLE_CATALOG	def
+TABLE_SCHEMA	information_schema
+TABLE_NAME	THREAD_POOL_WAITS
+TABLE_TYPE	SYSTEM VIEW
+ENGINE	NULL
+VERSION	10
+ROW_FORMAT	NULL
+TABLE_ROWS	#TBLR#
+AVG_ROW_LENGTH	#ARL#
+DATA_LENGTH	#DL#
+MAX_DATA_LENGTH	#MDL#
+INDEX_LENGTH	#IL#
+DATA_FREE	#DF#
+AUTO_INCREMENT	#AI#
+CREATE_TIME	#CRT#
+UPDATE_TIME	#UT#
+CHECK_TIME	#CT#
+TABLE_COLLATION	NULL
+CHECKSUM	NULL
+CREATE_OPTIONS	#CO#
+TABLE_COMMENT	#TC#
+user_comment	
+Separator	-----------------------------------------------------
+TABLE_CATALOG	def
+TABLE_SCHEMA	information_schema
 TABLE_NAME	TRIGGERS
 TABLE_TYPE	SYSTEM VIEW
 ENGINE	NULL
diff --git a/mysql-test/suite/information_schema/r/information_schema_db.result b/mysql-test/suite/information_schema/r/information_schema_db.result
index bf389e1de0a..680682fd426 100644
--- a/mysql-test/suite/information_schema/r/information_schema_db.result
+++ b/mysql-test/suite/information_schema/r/information_schema_db.result
@@ -48,6 +48,10 @@ TABLES_EXTENSIONS
 TABLE_CONSTRAINTS
 TABLE_CONSTRAINTS_EXTENSIONS
 TABLE_PRIVILEGES
+THREAD_POOL_GROUPS
+THREAD_POOL_QUEUES
+THREAD_POOL_STATS
+THREAD_POOL_WAITS
 TRIGGERS
 USER_ATTRIBUTES
 USER_PRIVILEGES
@@ -63,6 +67,10 @@ TABLES_EXTENSIONS
 TABLE_CONSTRAINTS
 TABLE_CONSTRAINTS_EXTENSIONS
 TABLE_PRIVILEGES
+THREAD_POOL_GROUPS
+THREAD_POOL_QUEUES
+THREAD_POOL_STATS
+THREAD_POOL_WAITS
 TRIGGERS
 create database `inf%`;
 create database mbase;
-- 
2.23.0

